front/components/sparkle/ThemeContext.tsx

```diff
@@ -9,52 +17,154 @@ interface ThemeContextType {
 }
 
 const DEFAULT_THEME: Theme = "system";
+const MEDIA = "(prefers-color-scheme: dark)";
 
 const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
 
-export function ThemeProvider({ children }: { children: React.ReactNode }) {
-  const [theme, setTheme] = useState<Theme>(DEFAULT_THEME);
-  const [isDark, setIsDark] = useState(false);
+function isTheme(value: string | null): value is Theme {
+  return (
+    typeof value === "string" && ["dark", "light", "system"].includes(value)
+  );
+}
 
-  useEffect(() => {
-    const savedTheme = localStorage.getItem("theme") as Theme;
-    if (savedTheme) {
-      setTheme(savedTheme);
+function getSavedTheme() {
+  if (typeof window === "undefined") {
+    return DEFAULT_THEME;
+  }
+
+  try {
+    const theme = localStorage.getItem("theme");
+
+    if (theme && isTheme(theme)) {
+      return theme;
     }
+
+    return DEFAULT_THEME;
+  } catch (e) {
+    // do nothing
+  }
+
+  return DEFAULT_THEME;
+}
+
+function getIsSystemDark() {
+  if (typeof window === "undefined") {
+    return false;
+  }
+
+  return window.matchMedia(MEDIA).matches;
+}
+
+// This is to disable animation temporaly when the theme is changed.
+const disableAnimation = () => {
+  const css = document.createElement("style");
+  css.appendChild(
+    document.createTextNode(
+      `*,*::before,*::after{-webkit-transition:none!important;-moz-transition:none!important;-o-transition:none!important;-ms-transition:none!important;transition:none!important}`
+    )
+  );
+  document.head.appendChild(css);
+
+  return () => {
+    // Force restyle
+    window.getComputedStyle(document.body);
+
+    // Wait for next animation frame before removing the disable transition CSS
+    requestAnimationFrame(() => {
+      document.head.removeChild(css);
+    });
+  };
+};
+
+// This is to avoid rendering the light theme first when the user has dark theme.
+// We want to run this before React hydration in the browser's global scope,
+// so we should not rely on any external variables.
+// TODO (05/12/2025 yuka) I'm not sure how to minify at build time, using manually minified script for now.
+const minifiedThemeScript = `function(){try{const theme=localStorage.getItem("theme")||"system";const isDark=theme==="dark"||(theme==="system"&&window.matchMedia("(prefers-color-scheme: dark)").matches);if(isDark){document.documentElement.classList.add("dark");document.documentElement.classList.add("s-dark")}}catch(e){}}`;
```

Comment:
It was not minified, and I'm not sure how to do it, so the easiest solution I could think of is manually minifying it 🙃